utils/auth/jwtHelpers.ts

```diff
@@ -56,25 +60,54 @@ export function withAnyUser<T>(
  * @returns NextApiHandler function.
  */
 export function withVerifiedUser<T>(
-  handler: (uid: string, req: NextApiRequest, res: NextApiResponse<T>) => void,
-): NextApiHandler<T> {
-  return async (req: NextApiRequest, res: NextApiResponse<T>) => {
+  handler: (uid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<T | EmptyPayload>>) => void,
+): NextApiHandler<ApiResponse<T | EmptyPayload>> {
+  return async (req: NextApiRequest, res: NextApiResponse<ApiResponse<T | EmptyPayload>>) => {
     let userDetails: UserDetailsFromRequest;
     try {
       userDetails = await getUserFromJwt(getBearerToken(req));
     } catch (e) {
-      return res.status(401).end(UNABLE_TO_AUTHENTICATE);
+      return res
+        .status(HTTP_STATUS_UNAUTHORIZED)
+        .json(createJsonResponse({}, [{ type: StatusMessageType.Error, message: UNABLE_TO_AUTHENTICATE }]));
     }
 
     const { uid, isAnonymous } = userDetails;
     if (isAnonymous) {
-      return res.status(403).end(USER_NOT_AUTHORIZED);
+      return res
+        .status(HTTP_STATUS_FORBIDDEN)
+        .json(createJsonResponse({}, [{ type: StatusMessageType.Error, message: USER_NOT_AUTHORIZED }]));
     }
 
     return handler(uid, req, res);
   };
 }
 
+/**
+ * Utility function that takes care of authenticating any user (anonymous / verified).
+ *
+ * Only calls the provided handler if the user has been successfully authenticated. Returns a 401 Unauthorized response
+ * if requirements are not met.
+ *
+ * @param handler Takes in an API handler function that needs an authenticated user.
+ * @returns NextApiHandler function.
+ */
+export function withAuthUser<T>(
```

Comment:
I'm confused because different things are named similarly.
```suggestion
export function withAuth<T>(
```